server/paymentProviders/stripe/payment-intent.ts

```diff
@@ -41,11 +47,94 @@ const processOrder = async (order: typeof models.Order): Promise<void> => {
+    // eslint-disable-next-line camelcase
+    paymentIntentParams.application_fee_amount = convertToStripeAmount(order.currency, applicationFee);
+  }
+
+  // eslint-disable-next-line camelcase
+  paymentIntentParams.payment_method_types = [order.paymentMethod?.type];
+  // eslint-disable-next-line camelcase
+  paymentIntentParams.payment_method = order.paymentMethod?.data?.stripePaymentMethodId;
+  paymentIntentParams.customer = order?.paymentMethod?.customerId;
+  paymentIntentParams.metadata = {
+    from: order.fromCollective ? `${config.host.website}/${order.fromCollective.slug}` : undefined,
+    to: `${config.host.website}/${order.collective.slug}`,
+    orderId: order.id,
+    chargeNumber: order.Subscription.chargeNumber,
+    chargeRetryCount: order.Subscription.chargeRetryCount,
+  };
+
+  try {
+    let paymentIntent = await stripe.paymentIntents.create(paymentIntentParams, {
+      stripeAccount: hostStripeAccount.username,
+    });
+
+    order.update({ data: { ...order.data, paymentIntent: { id: paymentIntent.id, status: paymentIntent.status } } });
+    await order.save();
+
+    paymentIntent = await stripe.paymentIntents.confirm(paymentIntent.id, {
+      stripeAccount: hostStripeAccount.username,
+    });
+
+    if (paymentIntent.next_action) {
+      const paymentIntentError = new Error('Payment Intent require action');
+      paymentIntentError['stripeAccount'] = hostStripeAccount.username;
+      paymentIntentError['stripeResponse'] = { paymentIntent };
+      throw paymentIntentError;
+    }
+
+    if (paymentIntent.status === 'processing') {
+      order.data = { ...order.data, paymentIntent };
+      order.update({ data: { ...order.data, paymentIntent } });
+      await order.save();
+      return;
+    } else if (paymentIntent.status !== 'succeeded') {
+      logger.error('Unknown error with Stripe Payment Intent.');
+      logger.error(paymentIntent);
+      reportMessageToSentry('Unknown error with Stripe Payment Intent', { extra: { paymentIntent } });
+      throw new Error('Something went wrong with the payment, please contact support@opencollective.com.');
+    }
+
+    // Recently, Stripe updated their library and removed the 'charges' property in favor of 'latest_charge',
+    // but this is something that only makes sense in the LatestApiVersion, and that's not the one we're using.
+    const charge = (paymentIntent as any).charges.data[0] as Stripe.Charge;
+    return createChargeTransactions(charge, { order });
+  } catch (e) {
+    const sanitizedError = pick(e, ['code', 'message', 'requestId', 'statusCode']);
+    const errorMessage = `Error processing Stripe Payment Intent: ${e.message}`;
+    logger.error(errorMessage, sanitizedError);
+    // Hard-delete the order so we can re-use the existing Payment Intent in the next attempt.
+    throw new Error(errorMessage);
+  }
+}
+
+const processOrder = async (order: typeof models.Order): Promise<void> => {
```

Comment:
The return type is misleading since `processRecurringOrder` can return a transaction.
We need to make sure this doesn't return anything, the existing logic is counting on this processor not creating any transactions:
https://github.com/opencollective/opencollective-api/blob/main/server/lib/payments.js#L577-L578